Lib/pathlib.py

```diff
@@ -407,6 +265,33 @@ def __reduce__(self):
         # when pickling related paths.
         return (self.__class__, tuple(self._parts))
 
+    @classmethod
+    def _split_root(cls, part):
+        sep = cls._flavour.sep
+        rel = cls._flavour.splitdrive(part)[1].lstrip(sep)
+        anchor = part[:len(part)-len(rel)]
```

Comment:
```suggestion
        anchor = part[:-len(rel)]
```

Reply:
Not equivalent when `rel` is empty, as you get `part[:-0]`, which is `part[:0]`

Reply:
Just realised I can use `removesuffix()` here, d'oh.